x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -476,12 +479,21 @@ public void testRemoteAccessAuthentication() throws IOException {
         );
         assertThat(
             authentication.getAuthenticatingSubject().getMetadata(),
-            hasEntry(AuthenticationField.REMOTE_ACCESS_AUTHENTICATION_KEY, remoteAccessAuthentication.getAuthentication().encode())
+            hasEntry(AuthenticationField.REMOTE_ACCESS_AUTHENTICATION_KEY, remoteAccessAuthentication.getAuthentication())
         );
         assertThat(
             authentication.getAuthenticatingSubject().getMetadata(),
             hasEntry(AuthenticationField.REMOTE_ACCESS_ROLE_DESCRIPTORS_KEY, remoteAccessAuthentication.getRoleDescriptorsBytesList())
         );
+
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            authentication.writeTo(out);
+            final StreamInput in = out.bytes().streamInput();
+            final Authentication deserialized = new Authentication(in);
+            assertThat(deserialized, equalTo(authentication));
+            ((List<?>) deserialized.getAuthenticatingSubject().getMetadata().get(AuthenticationField.REMOTE_ACCESS_ROLE_DESCRIPTORS_KEY))
+                .forEach(item -> assertThat(item, isA(RoleDescriptorsBytes.class)));
```

Comment:
Assert the type because `RoleDescriptorsBytes.equals(BytesReference)` can be `true`.

Reply:
Just noticed that you are adding an `equals` method to the subclass (#93662). So this assertion is no longer necessary. Removing.